neo/Network/P2P/Payloads/WitnessScope.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Neo.Network.P2P.Payloads
+{
+    public enum WitnessScope : byte
+    {
+        /// <summary>
+        /// (neo2) - no params
+        /// </summary>
+        Global = 0x00,
+
+        /// <summary>
+        /// EntryOnly means that this condition must hold: EntryScriptHash == CallingScriptHash
+        /// No params is needed, as the witness/permission/signature given on first invocation will automatically expire if entering deeper internal invokes
+        /// This can be default safe choice for native NEO/GAS (previously used on Neo 2 as "attach" mode)
+        /// </summary>
+        EntryOnly = 0x01,
+
+        /// <summary>
+        /// Custom hash for contract-specific
+        /// </summary>
+        CustomScriptHash = 0x02,
```

Comment:
What should I do if I need to authorize two script hashes?

Reply:
In that case you would require two witnesses on header... that's a necessary case to be studied.
Because on validation, we would need to know that there are two hash scopes, and if it satisfies one of them would be enough. 99% of cases are solved already with EntryOnly, but this was designed as an "extra" security, or when you cannot fit into EntryOnly (you send your script for another contract to execute your order, similar to DEX).
If we allow multiple CosignerUsage with same hash, I don't see much bad things happening, as user will pay for the price anyway.
I only see this happening in a atomic swap situation, is that your thoughts too?

Reply:
@erikzhang there are tests now for multiple cosigners. Feature works fine.